qiskit/algorithms/optimizers/umda.py

```diff
@@ -0,0 +1,373 @@
+        generation can sample from it. Update the vector of normal distributions
+        """
+
+        for i in range(self._n_variables):
+            self._vector[0, i], self._vector[1, i] = norm.fit(self._generation[:, i])
+            if self._vector[1, i] < self.STD_BOUND:
+                self._vector[1, i] = self.STD_BOUND
+
+    def minimize(
+        self,
+        fun: Callable[[POINT], float],
+        x0: POINT,
+        jac: Optional[Callable[[POINT], POINT]] = None,
+        bounds: Optional[List[Tuple[float, float]]] = None,
+    ) -> OptimizerResult:
+
+        not_better_count = 0
+        result = OptimizerResult()
+
+        self._n_variables = len(x0)
+        self._best_cost_global = 999999999999
+        self._best_ind_global = 9999999
+        history = []
+        self._evaluations = np.array(0)
+
+        self._vector = self._initialization()
+
+        # initialization of generation
+        self._generation = algorithm_globals.random.normal(
+            self._vector[0, :], self._vector[1, :], [self._size_gen, self._n_variables]
+        )
+
+        for _ in range(self._max_iter):
+            self._check_generation(fun)
+            self._truncation()
+            self._update_vector()
+
+            best_mae_local = min(self._evaluations)
+
+            history.append(best_mae_local)
+            best_ind_local = np.where(self._evaluations == best_mae_local)[0][0]
+            best_ind_local = self._generation[best_ind_local]
+
+            # update the best values ever
+            if best_mae_local < self._best_cost_global:
+                self._best_cost_global = best_mae_local
+                self._best_ind_global = best_ind_local
+                not_better_count = 0
+
+            else:
+                not_better_count += 1
+                if not_better_count == self._dead_iter:
+                    break
+
+            self._new_generation()
+
+        result.x = self._best_ind_global
+        result.fun = self._best_cost_global
+        result.nfev = len(history) * self._size_gen
+
+        return result
+
+    @property
+    def size_gen(self) -> int:
+        """Returns the size of the generations (number of individuals per generation)"""
+        return self._size_gen
+
+    @size_gen.setter
+    def size_gen(self, value: int):
+        """
+        Sets the size of the generations of the algorithm.
+
+        Args:
+            value: Size of the generations (number of individuals per generation).
+
+        Raises:
+            ValueError: If `value` is lower than 1.
+        """
+        if value <= 0:
+            raise ValueError("The size of the generation should be greater than 0.")
+        self._size_gen = value
+
+    @property
+    def max_iter(self) -> int:
+        """Returns the maximum number of iterations"""
+        return self._max_iter
+
+    @max_iter.setter
+    def max_iter(self, value: int):
```

Comment:
As the parameter on the constructor is `maxiter`, without an underscore, I think it would be better if this is the same so they more directly relate to  one another - ie. `def maxiter(self) -> int:` If you want the internal variable with an underscore thats fine, the end-user of the public API will never see this.